### ISSUE TITLE
Follow XDG Base directory specs

### ISSUE THREAD

**Issue by MckinleyX** [leanprover/elan#75]
Hi!
On Linux, elan stores files in `~/.elan`.

It may be better to follow the [XDG desktop specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html). 

In particular,
Configuration files should go under `$XDG_CONFIG_HOME` (if empty, `$HOME/.config/`)
Data files should go under `$XDG_DATA_HOME` (if empty, `$HOME/.local/share/`)

**Comment by sbatial** [comment]
I agree, that the XDG standard should probably be the default, but it is already possible to set ELAN_HOME.

**Comment by bigodel** [comment]
Bumping this up. Indeed you can set `ELAN_HOME` (where is that documented? Only found out about this through this issue) but it's not just about that, it's about using `XDG_BIN_HOME` to install binaries, for example

**Comment by Vinfall** [comment]
As elan-init.sh is edited from rust, I guess `$ELAN_HOME` also inherited the problematic behavior from `$CARGO_HOME`. That's in no way XDG compliant, actually, as data and config are both stored in `${ELAN,CARGO}_HOME`.

The correct way to do it is:
- place `settings.toml` in `$XDG_CONFIG_HOME/elan` (by default `$HOME/.config/elan`)
- place `toolchains/*` in `$XDG_DATA_HOME/elan` (by default `$HOME/.local/share/elan`)
- place `tmp/*` in `$XDG_CACHE_HOME/elan` (by default `$HOME/.cache/elan`)
- place `bin/*` in `$HOME/.local/bin`
- remove `env` and suggest it in command line prompt directly

PS: There is no such thing as `$XDG_BIN_HOME`, [the spec](https://specifications.freedesktop.org/basedir/latest/#variables) only suggests the use of `$HOME/.local/bin` 

>  User-specific executable files may be stored in `$HOME`/.local/bin. Distributions should ensure this directory shows up in the UNIX `$PATH` environment variable, at an appropriate place.
